controllers/windowsmachine/windowsmachine_controller.go

```diff
@@ -425,7 +426,7 @@ func (r *ReconcileWindowsMachine) deleteMachine(machine *mapi.Machine) error {
 // addWorkerNode configures the given Windows VM, adding it as a node object to the cluster
 func (r *ReconcileWindowsMachine) addWorkerNode(ipAddress, instanceID, machineName string, platform oconfig.PlatformType) error {
 	nc, err := nodeconfig.NewNodeConfig(r.k8sclientset, ipAddress, instanceID, machineName, r.clusterServiceCIDR,
-		r.vxlanPort, r.signer, platform)
+		r.vxlanPort, r.signer, platform, r.log)
```

Comment:
This is a general question that applies everywhere: should we be re-using the logger here? Shouldn't this be an internal field that the caller shouldn't be exposed to? IOW should we passing loggers around or be instantiating them in the New* functions.